### PR TITLE
Issue #836 Make MyPy stricter by enabling follow_imports

### LINKED ISSUE #836: Make MyPy stricter. Enable follow_imports

### PR DESCRIPTION
**Pull request by Manangka**
Fixes #836

In this commit the settings of MyPy are made stricter by enabling the follow_imports flag. By enabling this flag, information about type hints from modules being imported are also taken into account when performing the static code analyzes

In the pipeline a threshold has been added to the MyPy step which tolerates 2 errors. These 2 errors have to do with mypy errors with regard to the regridder. Those errors are better handled in a separate story. Once resolved the threshold can be set to 0

# Checklist

- [x] Links to correct issue
- [ ] Update changelog, if changes affect users
- [x] PR title starts with ``Issue #nr``, e.g. ``Issue #737``
- [ ] Unit tests were added
- [ ] **If feature added**: Added/extended example

### CODE REVIEW

File: imod/mf6/npf.py [hunk 1]
```diff
@@ -11,6 +11,18 @@
...
+
+def _dataarray_to_bool(griddataarray: GridDataArray) -> bool:
+    if griddataarray is None or griddataarray.values is None:
+        return False
+
+    if griddataarray.values.size != 1:
+        raise ValueError(
+            "Dataarray is not a single value. Unable to convert to boolean"
+        )
+    return griddataarray.values.item()
```
Comment (luitjansl) [inline imod/mf6/npf.py:25]: you could add a check on the dtype of griddataarray. It seems like in the use cases, it should be an actual bool and not an integer or float (and if it is, there may be something wrong)

**Review by luitjansl — APPROVED** [review]
looks good!

File: imod/mf6/npf.py (hunk 1 above)
Comment (Manangka) [reply imod/mf6/npf.py:31]: Good point. Added the check

**Comment by JoerivanEngelen** [comment]
Related to: https://github.com/Deltares/imod-python/issues/563

**Pull request merged**